core/ui/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/core/ui/exercise/ExerciseListItem.kt

```diff
@@ -79,16 +110,40 @@ private fun ExerciseTypeIcon(modifier: Modifier, exercise: Exercise) {
     Box(modifier = modifier) {
         Icon(
             modifier = Modifier
-                .fillMaxSize()
-                .padding(8.dp),
+                .fillMaxSize(),
             painter = getExerciseTypeIconPainter(exercise),
             contentDescription = null
         )
     }
 }
 
 /**
- * Displays the exercise title, the due data and the participation info. The participation info is automatically updated.
+ * Displays a rectangle next to the text to show the difficulty of the exercise.
+ */
+@Composable
+private fun DifficultyRectangle(difficulty: Exercise.Difficulty) {
+    Box(modifier = Modifier
+        .fillMaxHeight()
+        .height(40.dp)
+        .width(10.dp)
+        .clip(RectangleShape)
+        .background(
+            color = when (difficulty) {
+                Exercise.Difficulty.EASY ->
+                    Color(0xff28a745)
```

Comment:
Magic Number: Please do not hardcode color values. Also this ignores that the app can be displayed in both light and dark theme. 

Instead, you should define file level composable properties:
```kotlin
val myColor: Color
    @Composable get() = if(isSystemInDarkTheme()) 0xdarkthemehex else 0xlightThemeHex
```

Reply:
See this for an example: https://github.com/ls1intum/artemis-android/blob/cf66c7ca14c77e2430f0b992ca3c5d9f52569c27/core/ui/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/core/ui/alert/TextAlertDialog.kt#L12-L11

Reply:
Thanks for your feedback. I added the file level definitions, however I did not specify any values for the dark theme though as the difficulty colors stay consistent between these themes for the iOS app as well.

Reply:
You can find the dark theme and light theme values in the theme definitions of the webapp. Just check the scss files of the webapp.

You can also text search this repository, most likely I have already used that color before and defined a light and dark theme value for it. In that case you could reuse it.

Reply:
One of them already exists: 
https://github.com/ls1intum/artemis-android/blob/ace82dcaaa78e9f3489aae062427a5d683718162/feature/quiz/src/main/kotlin/de/tum/informatics/www1/artemis/native_app/feature/quiz/participation/QuizParticipationScreen.kt#L52

It would be easiest to just create a new file in the core/ui module de.tum.informatics.www1.artemis.native_app.core.ui.material package and put the color properties in there. Then we can reuse them among the project.

Reply:
Maybe, as part of this PR, just create the file in the core/ui module and put the colors in there to share it among the project. We can add the dark theme values in a future PR.

Reply:
I just added a new file for the difficulty colors and opened [this issue](https://github.com/ls1intum/artemis-android/issues/48) to address further changes to the color management.